app/(app)/settings/page.tsx

```diff
@@ -30,13 +30,32 @@ export default async function Page() {
     redirect("/get-started");
   }
 
-  const user = await prisma.user.findUnique({
+  let user = await prisma.user.findUnique({
```

Comment:
Any reason this is a let now?

Reply:
If the username string contains an accent we reassign user variable to the updated user on line 47.
> user = await prisma.user.update({

This updates user is then passed on to the settings page. 


Also coderabbits suggestion
> Error Handling: There's no explicit error handling around the database update operation. 

Should I wrap the update in a try.. catch?
The other prisma update in this component does not have one.

Reply:
Aha! I get why I'm confused. 

The change for the user cleanup should be in this block:

```ts
 if (!user?.username) {
    const nanoid = customAlphabet("1234567890ABCDEFGHIJKLMNOPQRSTUVWXYZ", 3);
    const initialUsername = `${(session.user.name || "").replace(
      /\s/g,
      "-",
    )}-${nanoid()}`.toLowerCase();

    const newUser = await prisma.user.update({
      where: {
        id: session.user.id,
      },
      data: {
        username: initialUsername,
      },
      select,
    });

    return <Content profile={newUser} />;
  }
```

This is where the initial username is set. This is where the bug slips in (not above; that's just fetching the profile.) Your schema should stop updating the profile to a username with special characters then (which it already does).

Reply:
Ah... right. I also now get why I am confused :laughing: 

I was under the mistaken impression that the username field was coming for the GitHub account. Which is why I was asking this earlier.
>My GitHub username is "john-paul-larkin"
However, when I am testing this the username field on account creation is always null. I am not sure why?

I will fix it now.